server/src/main/java/org/opensearch/autotagging/AutoTaggingRegistry.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.autotagging;
+
+import org.opensearch.ResourceNotFoundException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *  Registry for managing auto-tagging attributes and feature types.
+ *  This class provides functionality to register and retrieve {@link Attribute} and {@link FeatureType} instances
+ *  used for auto-tagging.
+ *
+ * @opensearch.experimental
+ */
+public class AutoTaggingRegistry {
+    public static final Map<String, FeatureType> featureTypesRegistryMap = new HashMap<>();
+
+    public static void registerFeatureType(FeatureType featureType) {
+        if (featureType == null || featureType.getName() == null) {
```

Comment:
Is there any limitations on the characters allowed in the featureType name? Also, should validate for non-empty featureType?